File: Dataset/Leetcode/test/7/196.py
class Solution:
    def XXX(self, x: int) -> int:
        if x < 0 :#将负数的转换成正数再进行翻转然后再转为负数
            c = 0
            c = c - x
            a = int(str(c)[::-1])
            a = -a
        else:    #正数直接翻转
            a = int(str(x)[::-1])
        if a < -(2**31) or a > 2**31 :#大小溢出返回0
            return 0
        return a

File: Dataset/Leetcode/train/98/196.cpp
 class Solution {
public:
    bool XXX(TreeNode* root) {
        if( root ){
            bool bl = XXX(root->left);
            cur = root;
            if( pre && cur && pre->val >= cur->val )
                return false;
            pre = cur;
            bool br = XXX(root->right);
            return bl && br;
        }
        return true;       
    }

    TreeNode* pre = nullptr;
    TreeNode* cur = nullptr;
};

